front-end/src/components/Reactionbar/ReactionButton.tsx

```diff
@@ -0,0 +1,144 @@
+	ReferendumPostAndCommentsQueryVariables,
+	MotionPostAndCommentsQueryVariables,
+	TreasuryProposalPostAndCommentsQueryVariables,
+
+	DiscussionPostAndCommentsQuery,
+	ProposalPostAndCommentsQuery,
+	ReferendumPostAndCommentsQuery,
+	MotionPostAndCommentsQuery,
+	TreasuryProposalPostAndCommentsQuery
+} from '../../generated/graphql';
+
+export interface ReactionButtonProps {
+	className?: string
+	reactionId: number
+	reaction: string
+	count: number
+	userMap: { [ key: string ]: string; }
+	postId?: number
+	commentId?: string
+	refetch?: (variables?:
+		ReferendumPostAndCommentsQueryVariables |
+		DiscussionPostAndCommentsQueryVariables |
+		ProposalPostAndCommentsQueryVariables |
+		MotionPostAndCommentsQueryVariables |
+		TreasuryProposalPostAndCommentsQueryVariables |
+		undefined) =>
+		Promise<ApolloQueryResult<TreasuryProposalPostAndCommentsQuery>> |
+		Promise<ApolloQueryResult<MotionPostAndCommentsQuery>> |
+		Promise<ApolloQueryResult<ReferendumPostAndCommentsQuery>> |
+		Promise<ApolloQueryResult<ProposalPostAndCommentsQuery>> |
+		Promise<ApolloQueryResult<DiscussionPostAndCommentsQuery>>
+}
+
+const ReactionButton = function ({
+	className,
+	reactionId,
+	reaction,
+	count,
+	userMap,
+	postId,
+	commentId,
+	refetch
+}: ReactionButtonProps) {
+	const { id } = useContext(UserDetailsContext);
+	const [addPostReactionMutation] = useAddPostReactionMutation();
+	const [addCommentReactionMutation] = useAddCommentReactionMutation();
+	const [deletePostReactionMutation] = useDeletePostReactionMutation();
+	const [deleteCommentReactionMutation] = useDeleteCommentReactionMutation();
+	const { queueNotification } = useContext(NotificationContext);
+	const reacted = !!userMap[`${id}`];
+
+	const handleReact = () => {
+		if (!id) {
+			queueNotification({
+				header: 'Failed!',
+				message: 'Please sign in to react',
+				status: NotificationStatus.ERROR
+			});
+			return;
+		}
+
+		if (postId) {
+			if (reacted) {
+				deletePostReactionMutation({
+					variables: {
+						postId,
+						reactionId,
+						userId: id
+					}
+				})
+					.catch((e) => console.error('Error in reacting to content',e));
+			} else {
+				addPostReactionMutation({
+					variables: {
+						postId,
+						reactionId,
+						userId: id
+					}
+				})
+					.catch((e) => console.error('Error in reacting to content',e));
+			}
+		}
+
+		if (commentId) {
+			if (reacted) {
+				deleteCommentReactionMutation({
+					variables: {
+						commentId,
+						reactionId,
+						userId: id
+					}
+				})
+					.catch((e) => console.error('Error in reacting to content',e));
+			} else {
+				addCommentReactionMutation({
+					variables: {
+						commentId,
+						reactionId,
+						userId: id
+					}
+				})
+					.catch((e) => console.error('Error in reacting to content',e));
+			}
+		}
+
+		refetch && refetch();
+	};
+
+	return (
+		<>
+			<Button
+				className={className + ' social'}
+				onClick={handleReact}
+			>
+				{reaction} {count}
```

Comment:
Didn't we want to show the users who reacted? username are not used afaics.

Reply:
Popup not working

Reply:
https://github.com/paritytech/polkassembly/issues/568